### ISSUE TITLE
The losses rec, kl both becomes NAN

### ISSUE THREAD

**Issue by blgpb** [orybkin/sigma-vae-pytorch#10]
Dear Prof. Rybkin,

I have read your paper "Simple and Effective VAE Training with Calibrated Decoders". It is a very nice work! I have learned a lot from it. I have replaced it with VAE in my work. However,  the losses of rec, kl are both becoming NAN. (https://github.com/orybkin/sigma-vae-pytorch/blob/master/model.py#L145)  Can you help me figure out this problem?

![image](https://github.com/orybkin/sigma-vae-pytorch/assets/40389895/04c8c31a-f6c9-4234-bcac-37838fbfe6f0)
![image](https://github.com/orybkin/sigma-vae-pytorch/assets/40389895/bfe4a85d-ccb1-456e-ae21-2da3fce6da00)
![image](https://github.com/orybkin/sigma-vae-pytorch/assets/40389895/c33acd5b-20bb-47ed-a082-f4afa6c53b36)
![image](https://github.com/orybkin/sigma-vae-pytorch/assets/40389895/f59b0bc3-ea7b-4382-96d4-a1b69afc65a6)

**Comment by orybkin** [comment]
Not a prof :) Are you able to run the code I provided on the data I tried
and replicate the results?

**Comment by blgpb** [comment]
Thank you for your reply!
I have sent you a detailed email. I look forward to your reply! Thank you very much!

**Comment by orybkin** [comment]
Note that you need to use the soft clipping to prevent the variance from
going to 0.

**Comment by blgpb** [comment]
Thank you very much. I have used the soft clipping to prevent the variance from going to 0 (https://github.com/orybkin/sigma-vae-pytorch/blob/master/model.py#L125C10-L125C10). Unfortunately, the problem still exists.

**Comment by orybkin** [comment]
I see. In that case I am not quite sure. I would check what the variance is
to make sure it's not too low. You can try to increase the lower limit. if
a normal VAE worked for you, you could try to make the implementation as
similar as possible in terms of loss scale, e.g. divide the loss by data
shape manually. If you are using half precision, you would need to be more
careful about loss scale. If this fails, I am not sure how to help, you'd
just have to debug it.
